src/hdmf/common/resources.py

```diff
@@ -693,46 +764,68 @@ def to_norm_tsv(self, **kwargs):
+
+        for i in range(len(self.children)):
+            df = self.children[i].to_dataframe()
+            df.to_csv(files[i], sep='\t', index=False)
+
+        with zipfile.ZipFile('er.zip', 'w') as zipF:
+          for file in files:
+              zipF.write(file)
+
+        # remove tsv files
+        for file in files:
+            os.remove(file)
 
     @classmethod
     @docval({'name': 'path', 'type': str, 'doc': 'path of the folder containing the tsv files to read'},
             returns="ExternalResources loaded from TSV", rtype="ExternalResources")
     def from_norm_tsv(cls, **kwargs):
         path = kwargs['path']
+        with zipfile.ZipFile(path+'/er.zip', 'r') as zip:
+            zip.extractall(path)
         tsv_paths = glob(path+'/*')
 
         for file in tsv_paths:
             file_name = os.path.basename(file)
             if file_name == 'files.tsv':
                 files_df = pd.read_csv(file, sep='\t').replace(np.nan, '')
                 files = FileTable().from_dataframe(df=files_df, name='files', extra_ok=False)
+                os.remove(file)
                 continue
             if file_name == 'keys.tsv':
                 keys_df = pd.read_csv(file, sep='\t').replace(np.nan, '')
                 keys = KeyTable().from_dataframe(df=keys_df, name='keys', extra_ok=False)
+                os.remove(file)
                 continue
             if file_name == 'entities.tsv':
                 entities_df = pd.read_csv(file, sep='\t').replace(np.nan, '')
                 entities = EntityTable().from_dataframe(df=entities_df, name='entities', extra_ok=False)
+                os.remove(file)
                 continue
             if file_name == 'objects.tsv':
                 objects_df = pd.read_csv(file, sep='\t').replace(np.nan, '')
                 objects = ObjectTable().from_dataframe(df=objects_df, name='objects', extra_ok=False)
+                os.remove(file)
                 continue
             if file_name == 'object_keys.tsv':
                 object_keys_df = pd.read_csv(file, sep='\t').replace(np.nan, '')
                 object_keys = ObjectKeyTable().from_dataframe(df=object_keys_df, name='object_keys', extra_ok=False)
+                os.remove(file)
+                continue
+            if file_name == 'entity_keys.tsv':
+                ent_key_df = pd.read_csv(file, sep='\t').replace(np.nan, '')
+                entity_keys = EntityKeyTable().from_dataframe(df=ent_key_df, name='entity_keys', extra_ok=False)
+                os.remove(file)
                 continue
 
         # we need to check the idx columns in entities, objects, and object_keys
-        keys_idx = entities['keys_idx']
-        for idx in keys_idx:
-            if not int(idx) < keys.__len__():
-                msg = "Key Index out of range in EntityTable. Please check for alterations."
+        entity_idx = entity_keys['entities_idx']
+        for idx in entity_idx:
+            if not int(idx) < entities.__len__():
```

Comment:
```suggestion
            if not int(idx) < len(entities):
```
It is much more common to call `len(x)` than `x.__len__()` (even though len(x) calls `x.__len__()`). Following conventions will help future maintainers understand the code. Deviating from convention suggests that there is some reason `len` is not being called.